### ISSUE TITLE
Release 4.9.0 - Alpha 3 - E2E UX tests - Amazon Security Hub

### ISSUE THREAD

**Issue by davidjiglesias** [wazuh/wazuh#24869]
## End-to-End (E2E) Testing Guideline

- **Documentation:** Always consult the development documentation for the current stage tag at [this link](https://documentation-dev.wazuh.com/v4.9.0-alpha3/index.html). Be careful because some of the description steps might refer to a current version in production, always navigate using the current development documention for the stage under test. Also, visit the following [pre-release package guide](https://docs.google.com/document/d/1fW8koy8P8OMvcxzIOXWGDw9qmKlju-XuRHD-Bdv3uP4/edit) to understand how to modify certain links and urls for the correct testing of the development packages.
- **Test Requirements:** Ensure your test comprehensively includes a full stack and agent/s deployment as per the Deployment requirements, detailing the machine OS, installed version, and revision.
- **Deployment Options:** While deployments can be local (using VMs, Vagrant, etc) or on the aws-dev account, opt for local deployments when feasible. For AWS access, coordinate with the DevOps team through [this link](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **External Accounts:** If tests require third-party accounts (e.g., GitHub, Azure, AWS, GCP), request the necessary access through the DevOps team [here](https://github.com/wazuh/internal-devel-requests/issues/new?assignees=&labels=level%2Ftask%2C+request%2Foperational%2C+type%2Fchange&projects=&template=operational--request.md&title=%3CTitle%3E).
- **Alerts:** Every test should generate a minimum of one end-to-end alert, from the agent to the dashboard, irrespective of test type.
- **Multi-node Testing:** For multi-node wazuh-manager tests, ensure agents are connected to both workers and the master node.
- **Package Verification:** Use the pre-release package that matches the current TAG you're testing. Confirm its version and revision.
- **Filebeat Errors:** If you encounter errors with Filebeat during testing, refer to [this Slack discussion](https://wazuh-team.slack.com/archives/C03BDG0K6JC/p1672168163537809) for insights and resolutions.
- **Known Issues:** Familiarize yourself with previously reported issues in the Known Issues section. This helps in identifying already recognized errors during testing.
- **Reporting New Issues:** Any new errors discovered during testing that aren't listed under Known Issues should be reported. Assign the issue to the corresponding team (QA if unsure), add the `Release testing` objective and `Urgent` priority. Communicate these to the team and QA via the c-release Slack channel.
- **Test Conduct:** It's imperative to be thorough in your testing, offering enough detail for reviewers. Incomplete tests might necessitate a redo.
- **Documentation Feedback:** Encountering documentation gaps, unclear guidelines, or anything that disrupts the testing or UX? Open an issue, especially if it's not listed under Known Issues. Please answer the feedback section, this is a mandatory step.
- **Format:** If this is your first time doing this, refer to the format (but not necessarily the content, as it may vary) of previous E2E tests, here you have an example https://github.com/wazuh/wazuh/issues/13994.
- **Status and completion:** Change the issue status within your team project accordingly. Once you finish testing and write the conclusions, move it to Pending review and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ). Beware that the reviewers might request additional information or task repetitions.
- **For reviewers:** Please move the issue to Pending final review and notify via Slack using the same thread if everything is ok, otherwise, perform an issue update with the requested changes and move it to On hold, increase the review_cycles in the  team project by one and notify the issue assignee via Slack using the same thread.

For the conclusions and the issue testing and updates, use the following legend:

**Status legend**
- 🟢 All checks passed
- 🟡 Found a known issue
- 🔴 Found a new error

## Issue delivery and completion
- **Initial delivery:** The issue's assignee must complete the testing and deliver the results by Jul 24, 2024 and notify the @wazuh/devel-pyserver team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ)
- **Review:** The @wazuh/devel-pyserver team will assign a reviewer and add it to the `review_assignee` field in the  project. The reviewer must then review the test steps and results. Ensure that all iteration cycles are completed by Jul 25, 2024 date (issue must be in `Pending final review` status) and notify the QA team via Slack using the [c-release channel](https://wazuh-team.slack.com/archives/C02A737S5MJ).
- **Auditor:** The QA team must audit, validate the results, and close the issue by Jul 26, 2024.

## Deployment requirements

| Component | Installation | Type | OS |
|----------|--------------|------|----|
| Indexer | [Quickstart](https://documentation-dev.wazuh.com/v4.9.0-alpha3/quickstart.html) | Single Node | Ubuntu 22.04 x86_64 |
| Server | Same as Indexer, all-in-one | Single node | Same as Indexer |
| Dashboard | Same as indexer, all-in-one | - | Same as Indexer |
| Agent | [Installing Wazuh agents](https://documentation-dev.wazuh.com/v4.9.0-alpha3/installation-guide/wazuh-agent/index.html) | - | Debian 12 x86_64 |

## Test description

- Configure AWS Security Hub in a Wazuh Manager and a Wazuh Agent. Follow the documentation page  https://documentation-dev.wazuh.com/v4.9.0-alpha3/cloud-security/amazon/services/supported-services/security-hub.html
- Generate Amazon Security Hub alerts. Check that generated alerts appear on the dashboard.
- Use [discard_regex option](https://documentation-dev.wazuh.com/v4.9.0-alpha3/user-manual/reference/ossec-conf/wodle-s3.html#bucket-discard-regex) to effectively discard certain types of events. 

## Known issues

There are no known issues.

## Conclusions

Summarize the errors detected (Known Issues included). Illustrate using the table below. **REMOVE CURRENT EXAMPLES**:

| **Status**     | **Test**    | **Failure type**    | **Notes**      |
|----------------|-------------|---------------------|----------------|
| 🟢 | Wazuh Manager Install |  |   |
| 🟢 | Wazuh Agent install  |  |   |
| 🔴 | AWS Security Hub Configuration | Complexity & Access rights | Documentation was too complex and referred to much external documentations, as well as it relayed user assumed knowledge on AWS features & concepts. [Issue 7588](https://github.com/wazuh/wazuh-documentation/issues/7588) |
| 🟢 | Wazuh Manager Configuration |  | Documentation was too complex and lack a simple example to build on top.
| 🟢 | Wazuh Manager Events & Dashboard|  | Events were already in the queue. Don't know how to generate them (or even if we have the rights to do it)
| 🔴 | Wazuh Manager discard regex | Test could not be completed | [Issue 7588](https://github.com/wazuh/wazuh-documentation/issues/7566)
| 🔴 | Wazuh Agent Configuration  | pyarrow module is required |  [Issue 7566](https://github.com/wazuh/wazuh-documentation/issues/7566)
| 🔴 | Wazuh Agent Events & Dashboard | The test could not be completed | Configuration didn't worked for the agent [Issue 7566](https://github.com/wazuh/wazuh-documentation/issues/7566)
| 🔴 | Wazuh Agent discard regex | The test could not be completed | Configuration didn't worked for the agent [Issue 7566](https://github.com/wazuh/wazuh-documentation/issues/7566)

## Feedback

We value your feedback. Please provide insights on your testing experience.

- Was the testing guideline clear? Were there any ambiguities?
  - It depends. If the guideline is targeted to an experience AWS administrator, it might be ok. But if the guideline is targeted to any Wazuh's user, then it should probably have more complete examples, and then giving references for more information, instead of just linking to a lot of Wazuh's and AWS's documentation links.
- Did you face any challenges not covered by the guideline?
  - I was not able to configure AWS in the timebox required by this E2E test, since there were a lot of AWS features & concepts that I am not familiar with and required too much reading and interpretation.
  - Also, I faced a lot of problems due to access restrictions on the provided AWS account, which had to be solved by DevOps team members whenever they appear, and it difficult a lot to know if things were being done properly
- Suggestions for improvement:
  - If the guideline is intended for all Wazuh's users, complete it with full examples on getting a simple example to work, and provide Wazuh's and AWS's documentation links as places to obtain further information or extend the example.
  - Provide for the E2E test procedure an already existing and configured account, or at least a profile or template ready to be required.
  - Whenever a new feature is implemented, there should be time dedicated to design and implemented the corresponding E2E by the developer
  - The first time a E2E is implemented, it might be a good idea that it is executed a developer's team member

## Reviewers validation

The criteria for completing this task is based on the validation of the conclusions and the test results by all reviewers.

All the checkboxes below must be marked in order to close this issue.
- [ ] @rauldpm
- [x] @wazuh/devel-pyserver

**Comment by ncvicchi** [comment]
## Environment install :green_circle: 

### Wazuh Server install :green_circle:

#### Resources:
VirtualBox VM, 8GB RAM, 8 cores, 100Gb disk
Ubuntu 22.04 fresh install, fully upgraded
Prerequisites: OpenSSH-server (for ease of use) and curl installed:

<details><summary>Prerequisites</summary>

```console
wazuh@wazuh-server:~$ sudo bash
[sudo] contraseña para wazuh: 
root@wazuh-server:/home/wazuh# apt-get update
Obj:1 http://ar.archive.ubuntu.com/ubuntu jammy InRelease
Obj:2 http://ar.archive.ubuntu.com/ubuntu jammy-updates InRelease               
Obj:3 http://ar.archive.ubuntu.com/ubuntu jammy-backports InRelease             
Des:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]
Descargados 129 kB en 1s (101 kB/s)
Leyendo lista de paquetes... Hecho
root@wazuh-server:/home/wazuh# apt-get upgrade
Leyendo lista de paquetes... Hecho
Creando árbol de dependencias... Hecho
Leyendo la información de estado... Hecho
Calculando la actualización... Hecho
Los paquetes indicados a continuación se instalaron de forma automática y ya no son necesarios.
  libwpe-1.0-1 libwpebackend-fdo-1.0-1
Utilice «sudo apt autoremove» para eliminarlos.
Los siguientes paquetes se han retenido:
  libfprint-2-2 python3-update-manager update-manager update-manager-core
0 actualizados, 0 nuevos se instalarán, 0 para eliminar y 4 no actualizados.
root@wazuh-server:/home/wazuh# apt install curl
Leyendo lista de paquetes... Hecho
Creando árbol de dependencias... Hecho
Leyendo la información de estado... Hecho
Los paquetes indicados a continuación se instalaron de forma automática y ya no son necesarios.
  libwpe-1.0-1 libwpebackend-fdo-1.0-1
Utilice «sudo apt autoremove» para eliminarlos.
Se instalarán los siguientes paquetes NUEVOS:
  curl
0 actualizados, 1 nuevos se instalarán, 0 para eliminar y 4 no actualizados.
Se necesita descargar 194 kB de archivos.
Se utilizarán 454 kB de espacio de disco adicional después de esta operación.
Des:1 http://ar.archive.ubuntu.com/ubuntu jammy-updates/main amd64 curl amd64 7.81.0-1ubuntu1.16 [194 kB]
Descargados 194 kB en 0s (2.540 kB/s)
Seleccionando el paquete curl previamente no seleccionado.
(Leyendo la base de datos ... 208765 ficheros o directorios instalados actualmente.)
Preparando para desempaquetar .../curl_7.81.0-1ubuntu1.16_amd64.deb ...
Desempaquetando curl (7.81.0-1ubuntu1.16) ...
Configurando curl (7.81.0-1ubuntu1.16) ...
Procesando disparadores para man-db (2.10.2-1) ...
root@wazuh-server:/home/wazuh# uname -a
Linux wazuh-server 6.5.0-44-generic #44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Jun 18 14:36:16 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```
</details>

Indexer, server and dashboard install
All 3 components installed by following the [Quickstart](https://documentation-dev.wazuh.com/v4.9.0-alpha3/quickstart.html) guide. URLs are modified to use pre-release packages instead of the documentation URLs:

<details><summary>Server install</summary>

```console
root@wazuh-server:/home/wazuh# curl -sO https://packages-dev.wazuh.com/4.9/wazuh-install.sh && sudo bash ./wazuh-install.sh -a
23/07/2024 20:25:47 INFO: Starting Wazuh installation assistant. Wazuh version: 4.9.0
23/07/2024 20:25:47 INFO: Verbose logging redirected to /var/log/wazuh-install.log
23/07/2024 20:25:50 INFO: --- Dependencies ----
23/07/2024 20:25:50 INFO: Installing gawk.
23/07/2024 20:25:53 INFO: Verifying that your system meets the recommended minimum hardware requirements.
23/07/2024 20:25:56 INFO: Wazuh web interface port will be 443.
23/07/2024 20:26:00 INFO: --- Dependencies ----
23/07/2024 20:26:00 INFO: Installing apt-transport-https.
23/07/2024 20:26:02 INFO: Installing debhelper.
23/07/2024 20:26:21 INFO: Wazuh development repository added.
23/07/2024 20:26:21 INFO: --- Configuration files ---
23/07/2024 20:26:21 INFO: Generating configuration files.
23/07/2024 20:26:21 INFO: Generating the root certificate.
23/07/2024 20:26:22 INFO: Generating Admin certificates.
23/07/2024 20:26:22 INFO: Generating Wazuh indexer certificates.
23/07/2024 20:26:22 INFO: Generating Filebeat certificates.
23/07/2024 20:26:22 INFO: Generating Wazuh dashboard certificates.
23/07/2024 20:26:22 INFO: Created wazuh-install-files.tar. It contains the Wazuh cluster key, certificates, and passwords necessary for installation.
23/07/2024 20:26:22 INFO: --- Wazuh indexer ---
23/07/2024 20:26:22 INFO: Starting Wazuh indexer installation.
23/07/2024 20:27:17 INFO: Wazuh indexer installation finished.
23/07/2024 20:27:18 INFO: Wazuh indexer post-install configuration finished.
23/07/2024 20:27:18 INFO: Starting service wazuh-indexer.
23/07/2024 20:27:25 INFO: wazuh-indexer service started.
23/07/2024 20:27:25 INFO: Initializing Wazuh indexer cluster security settings.
23/07/2024 20:27:27 INFO: Wazuh indexer cluster security configuration initialized.
23/07/2024 20:27:27 INFO: Wazuh indexer cluster initialized.
23/07/2024 20:27:27 INFO: --- Wazuh server ---
23/07/2024 20:27:27 INFO: Starting the Wazuh manager installation.
23/07/2024 20:28:39 INFO: Wazuh manager installation finished.
23/07/2024 20:28:39 INFO: Wazuh manager vulnerability detection configuration finished.
23/07/2024 20:28:39 INFO: Starting service wazuh-manager.
23/07/2024 20:28:53 INFO: wazuh-manager service started.
23/07/2024 20:28:53 INFO: Starting Filebeat installation.
23/07/2024 20:29:00 INFO: Filebeat installation finished.
23/07/2024 20:29:02 INFO: Filebeat post-install configuration finished.
23/07/2024 20:29:02 INFO: Starting service filebeat.
23/07/2024 20:29:02 INFO: filebeat service started.
23/07/2024 20:29:02 INFO: --- Wazuh dashboard ---
23/07/2024 20:29:02 INFO: Starting Wazuh dashboard installation.
23/07/2024 20:30:24 INFO: Wazuh dashboard installation finished.
23/07/2024 20:30:24 INFO: Wazuh dashboard post-install configuration finished.
23/07/2024 20:30:24 INFO: Starting service wazuh-dashboard.
23/07/2024 20:30:24 INFO: wazuh-dashboard service started.
23/07/2024 20:30:25 INFO: Updating the internal users.
23/07/2024 20:30:27 INFO: A backup of the internal users has been saved in the /etc/wazuh-indexer/internalusers-backup folder.
23/07/2024 20:30:32 INFO: The filebeat.yml file has been updated to use the Filebeat Keystore username and password.
23/07/2024 20:31:04 INFO: Initializing Wazuh dashboard web application.
23/07/2024 20:31:04 INFO: Wazuh dashboard web application initialized.
23/07/2024 20:31:04 INFO: --- Summary ---
23/07/2024 20:31:04 INFO: You can access the web interface https://<wazuh-dashboard-ip>:443
    User: admin
    Password: DBKiFfD168gC7mL.RCn.3XGNh7*kbCLO
23/07/2024 20:31:04 INFO: --- Dependencies ----
23/07/2024 20:31:04 INFO: Removing gawk.
23/07/2024 20:31:07 INFO: Installation finished.
root@wazuh-server:/home/wazuh# /var/ossec/bin/wazuh-control info
WAZUH_VERSION="v4.9.0"
WAZUH_REVISION="40903"
WAZUH_TYPE="server"

```
</details>

<details><summary>Wazuh's dashboard</summary>

![image](https://github.com/user-attachments/assets/54199725-6556-401f-9a45-b479c1a5a559)

</details>

### Wazuh Agent install :green_circle: 

#### Resources:
VirtualBox VM, 8GB RAM, 8 cores, 100Gb disk
Debian 12.6 fresh install, fully upgraded
Prerequisites: OpenSSH-server (for ease of use) and curl installed:

<details><summary>Prerequisites</summary>

```console

wazuh@wazuh-agent:~$ su root
Contraseña: 
root@wazuh-agent:/home/wazuh# apt install curl
Leyendo lista de paquetes... Hecho
Creando árbol de dependencias... Hecho
Leyendo la información de estado... Hecho
Se instalarán los siguientes paquetes NUEVOS:
  curl
0 actualizados, 1 nuevos se instalarán, 0 para eliminar y 0 no actualizados.
Se necesita descargar 314 kB de archivos.
Se utilizarán 500 kB de espacio de disco adicional después de esta operación.
Des:1 http://deb.debian.org/debian bookworm/main amd64 curl amd64 7.88.1-10+deb12u6 [314 kB]
Descargados 314 kB en 0s (8.791 kB/s)
Seleccionando el paquete curl previamente no seleccionado.
(Leyendo la base de datos ... 154809 ficheros o directorios instalados actualmente.)
Preparando para desempaquetar .../curl_7.88.1-10+deb12u6_amd64.deb ...
Desempaquetando curl (7.88.1-10+deb12u6) ...
Configurando curl (7.88.1-10+deb12u6) ...
Procesando disparadores para man-db (2.11.2-2) ...
root@wazuh-agent:/home/wazuh# apt-get update
Obj:1 http://deb.debian.org/debian bookworm InRelease
Obj:2 http://deb.debian.org/debian bookworm-updates InRelease
Obj:3 http://security.debian.org/debian-security bookworm-security InRelease
Leyendo lista de paquetes... Hecho
root@wazuh-agent:/home/wazuh# uname -a
Linux wazuh-agent 6.1.0-23-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.99-1 (2024-07-15) x86_64 GNU/Linux

```

</details>

<details><summary>Wazuh Agent install</summary>

```console
root@wazuh-agent:/home/wazuh# curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | gpg --no-default-keyring --keyring gnupg-ring:/usr/share/keyrings/wazuh.gpg --import && chmod 644 /usr/share/keyrings/wazuh.gpg
gpg: anillo '/usr/share/keyrings/wazuh.gpg' creado
gpg: creado el directorio '/root/.gnupg'
gpg: /root/.gnupg/trustdb.gpg: se ha creado base de datos de confianza
gpg: clave 96B3EE5F29111145: clave pública "Wazuh.com (Wazuh Signing Key) <support@wazuh.com>" importada
gpg: Cantidad total procesada: 1
gpg:               importadas: 1
root@wazuh-agent:/home/wazuh# echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages-dev.wazuh.com/pre-release/apt/ unstable main" | tee -a /etc/apt/sources.list.d/wazuh.list
deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages-dev.wazuh.com/pre-release/apt/ unstable main
root@wazuh-agent:/home/wazuh# apt-get update
Obj:1 http://security.debian.org/debian-security bookworm-security InRelease
Obj:2 http://deb.debian.org/debian bookworm InRelease
Obj:3 http://deb.debian.org/debian bookworm-updates InRelease
Des:4 https://packages-dev.wazuh.com/pre-release/apt unstable InRelease [17,3 kB]
Des:5 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 Packages [39,6 kB]
Descargados 56,9 kB en 2s (25,5 kB/s)
Leyendo lista de paquetes... Hecho
root@wazuh-agent:/home/wazuh# WAZUH_MANAGER="192.168.1.77" apt-get install wazuh-agent=4.9.0-1
Leyendo lista de paquetes... Hecho
Creando árbol de dependencias... Hecho
Leyendo la información de estado... Hecho
Se instalarán los siguientes paquetes NUEVOS:
  wazuh-agent
0 actualizados, 1 nuevos se instalarán, 0 para eliminar y 0 no actualizados.
Se necesita descargar 10,7 MB de archivos.
Se utilizarán 37,2 MB de espacio de disco adicional después de esta operación.
Des:1 https://packages-dev.wazuh.com/pre-release/apt unstable/main amd64 wazuh-agent amd64 4.9.0-1 [10,7 MB]
Descargados 10,7 MB en 3s (3.564 kB/s)  
Preconfigurando paquetes ...
Seleccionando el paquete wazuh-agent previamente no seleccionado.
(Leyendo la base de datos ... 154816 ficheros o directorios instalados actualmente.)
Preparando para desempaquetar .../wazuh-agent_4.9.0-1_amd64.deb ...
Desempaquetando wazuh-agent (4.9.0-1) ...
Configurando wazuh-agent (4.9.0-1) ...
root@wazuh-agent:/home/wazuh# systemctl daemon-reload
root@wazuh-agent:/home/wazuh# systemctl enable wazuh-agent
Created symlink /etc/systemd/system/multi-user.target.wants/wazuh-agent.service → /lib/systemd/system/wazuh-agent.service.
root@wazuh-agent:/home/wazuh# systemctl start wazuh-agent
root@wazuh-agent:/home/wazuh# sed -i "s/^deb/#deb/" /etc/apt/sources.list.d/wazuh.list
root@wazuh-agent:/home/wazuh# apt-get update
Obj:1 http://security.debian.org/debian-security bookworm-security InRelease
Obj:2 http://deb.debian.org/debian bookworm InRelease
Obj:3 http://deb.debian.org/debian bookworm-updates InRelease
Leyendo lista de paquetes... Hecho
root@wazuh-agent:/home/wazuh# /var/ossec/bin/wazuh-control info
WAZUH_VERSION="v4.9.0"
WAZUH_REVISION="40903"
WAZUH_TYPE="agent"
root@wazuh-agent:/home/wazuh# 
```

</details>

<details><summary>Agent enrollment (dashboard)</summary>

![image](https://github.com/user-attachments/assets/bcbb37b3-0deb-4de3-9bf6-55818d09b2ec)

</details>

<details><summary>Manager's log</summary>

```console
root@wazuh-server:/home/wazuh# cat /var/ossec/logs/ossec.log 
2024/07/23 20:28:41 wazuh-modulesd:router: INFO: Loaded router module.
2024/07/23 20:28:41 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
2024/07/23 20:28:43 wazuh-csyslogd: INFO: Remote syslog server not configured. Clean exit.
2024/07/23 20:28:43 wazuh-dbd: INFO: Database not configured. Clean exit.
2024/07/23 20:28:43 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
2024/07/23 20:28:43 wazuh-agentlessd: INFO: Not configured. Exiting.
2024/07/23 20:28:43 wazuh-authd: INFO: Started (pid: 55132).
2024/07/23 20:28:43 wazuh-authd: INFO: Accepting connections on port 1515. No password required.
2024/07/23 20:28:43 wazuh-authd: INFO: Setting network timeout to 1.000000 sec.
2024/07/23 20:28:44 wazuh-db: INFO: Started (pid: 55157).
2024/07/23 20:28:44 wazuh-db: INFO: Created Global database backup "backup/db/global.db-backup-2024-07-23-20:28:44.gz"
2024/07/23 20:28:45 wazuh-execd: INFO: Started (pid: 55182).
2024/07/23 20:28:46 wazuh-analysisd: INFO: Total rules enabled: '7006'
2024/07/23 20:28:46 wazuh-analysisd: INFO: Started (pid: 55200).
2024/07/23 20:28:46 wazuh-analysisd: INFO: EPS limit disabled
2024/07/23 20:28:46 wazuh-analysisd: INFO: (7200): Logtest started
2024/07/23 20:28:47 wazuh-syscheckd: INFO: Started (pid: 55297).
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/boot', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6000): Starting daemon...
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2024/07/23 20:28:47 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
2024/07/23 20:28:47 rootcheck: INFO: Starting rootcheck scan.
2024/07/23 20:28:48 wazuh-remoted: INFO: Started (pid: 55313). Listening on port 1514/TCP (secure).
2024/07/23 20:28:48 wazuh-remoted: INFO: (1410): Reading authentication keys file.
2024/07/23 20:28:49 wazuh-logcollector: INFO: Monitoring output of command(360): df -P
2024/07/23 20:28:49 wazuh-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
2024/07/23 20:28:49 wazuh-logcollector: INFO: Monitoring full output of command(360): last -n 20
2024/07/23 20:28:49 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
2024/07/23 20:28:49 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/log/dpkg.log'.
2024/07/23 20:28:49 wazuh-logcollector: INFO: Started (pid: 55348).
2024/07/23 20:28:49 wazuh-monitord: INFO: Started (pid: 55364).
2024/07/23 20:28:49 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2024/07/23 20:28:49 wazuh-syscheckd: INFO: FIM sync module started.
2024/07/23 20:28:50 wazuh-modulesd:router: INFO: Loaded router module.
2024/07/23 20:28:50 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
2024/07/23 20:28:50 wazuh-modulesd: INFO: Started (pid: 55401).
2024/07/23 20:28:50 wazuh-modulesd:agent-upgrade: INFO: (8153): Module Agent Upgrade started.
2024/07/23 20:28:50 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
2024/07/23 20:28:50 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
2024/07/23 20:28:50 wazuh-modulesd:vulnerability-scanner: INFO: Starting vulnerability_scanner module.
2024/07/23 20:28:50 sca: INFO: Module started.
2024/07/23 20:28:50 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:28:50 wazuh-modulesd:task-manager: INFO: (8200): Module Task Manager started.
2024/07/23 20:28:50 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/23 20:28:50 wazuh-modulesd:database: INFO: Module started.
2024/07/23 20:28:50 wazuh-modulesd:router: INFO: Starting router module.
2024/07/23 20:28:50 wazuh-modulesd:content_manager: INFO: Starting content_manager module.
2024/07/23 20:28:50 wazuh-modulesd:download: INFO: Module started.
2024/07/23 20:28:50 wazuh-modulesd:control: INFO: Starting control thread.
2024/07/23 20:28:50 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:28:50 wazuh-modulesd:syscollector: INFO: Module started.
2024/07/23 20:28:50 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/07/23 20:28:50 indexer-connector: WARNING: IndexerConnector initialization failed for index 'wazuh-states-vulnerabilities-wazuh-server', retrying until the connection is successful.
2024/07/23 20:28:50 wazuh-modulesd:vulnerability-scanner: INFO: Starting database file decompression.
2024/07/23 20:28:50 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2024/07/23 20:28:51 wazuh-logcollector: INFO: (9203): Monitoring journal entries.
2024/07/23 20:28:55 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:28:55 sca: INFO: Security Configuration Assessment scan finished. Duration: 5 seconds.
2024/07/23 20:29:05 indexer-connector: INFO: IndexerConnector initialized successfully for index: wazuh-states-vulnerabilities-wazuh-server.
2024/07/23 20:29:14 wazuh-modulesd:vulnerability-scanner: INFO: Database decompression finished.
2024/07/23 20:29:15 wazuh-modulesd:vulnerability-scanner: INFO: Vulnerability scanner module started.
2024/07/23 20:29:17 wazuh-modulesd:vulnerability-scanner: INFO: Initiating update feed process.
2024/07/23 20:29:39 rootcheck: INFO: Ending rootcheck scan.
2024/07/23 20:30:32 wazuh-modulesd:syscollector: INFO: Stop received for Syscollector.
2024/07/23 20:30:32 wazuh-modulesd:syscollector: INFO: Module finished.
2024/07/23 20:30:32 wazuh-modulesd:vulnerability-scanner: INFO: Stopping vulnerability_scanner module.
2024/07/23 20:30:33 wazuh-modulesd:vulnerability-scanner: INFO: Feed update interrupted: Module stopped..
2024/07/23 20:30:33 wazuh-modulesd:router: INFO: Stopping router module.
2024/07/23 20:30:33 wazuh-modulesd:content_manager: INFO: Stopping content_manager module.
2024/07/23 20:30:38 wazuh-monitord: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:38 wazuh-logcollector: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:38 wazuh-remoted: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:38 wazuh-syscheckd: INFO: (1756): Shutdown received. Releasing resources.
2024/07/23 20:30:39 wazuh-syscheckd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:39 wazuh-analysisd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:39 wazuh-execd: INFO: (1314): Shutdown received. Deleting responses.
2024/07/23 20:30:39 wazuh-execd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:39 wazuh-db: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:40 wazuh-db: INFO: Graceful process shutdown.
2024/07/23 20:30:40 wazuh-authd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:30:41 wazuh-authd: INFO: Exiting...
2024/07/23 20:30:42 wazuh-modulesd:router: INFO: Loaded router module.
2024/07/23 20:30:42 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
2024/07/23 20:30:44 wazuh-csyslogd: INFO: Remote syslog server not configured. Clean exit.
2024/07/23 20:30:44 wazuh-dbd: INFO: Database not configured. Clean exit.
2024/07/23 20:30:44 wazuh-integratord: INFO: Remote integrations not configured. Clean exit.
2024/07/23 20:30:44 wazuh-agentlessd: INFO: Not configured. Exiting.
2024/07/23 20:30:44 wazuh-authd: INFO: Started (pid: 59032).
2024/07/23 20:30:44 wazuh-authd: INFO: Accepting connections on port 1515. No password required.
2024/07/23 20:30:44 wazuh-authd: INFO: Setting network timeout to 1.000000 sec.
2024/07/23 20:30:45 wazuh-db: INFO: Started (pid: 59048).
2024/07/23 20:30:46 wazuh-execd: INFO: Started (pid: 59077).
2024/07/23 20:30:47 wazuh-analysisd: INFO: Total rules enabled: '7006'
2024/07/23 20:30:47 wazuh-analysisd: INFO: Started (pid: 59092).
2024/07/23 20:30:47 wazuh-analysisd: INFO: (7200): Logtest started
2024/07/23 20:30:47 wazuh-analysisd: INFO: EPS limit disabled
2024/07/23 20:30:48 wazuh-syscheckd: INFO: Started (pid: 59188).
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/boot', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6000): Starting daemon...
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2024/07/23 20:30:48 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
2024/07/23 20:30:48 rootcheck: INFO: Starting rootcheck scan.
2024/07/23 20:30:49 wazuh-remoted: INFO: Started (pid: 59205). Listening on port 1514/TCP (secure).
2024/07/23 20:30:49 wazuh-remoted: INFO: (1410): Reading authentication keys file.
2024/07/23 20:30:50 wazuh-logcollector: INFO: Monitoring output of command(360): df -P
2024/07/23 20:30:50 wazuh-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
2024/07/23 20:30:50 wazuh-logcollector: INFO: Monitoring full output of command(360): last -n 20
2024/07/23 20:30:50 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
2024/07/23 20:30:50 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/log/dpkg.log'.
2024/07/23 20:30:50 wazuh-logcollector: INFO: Started (pid: 59255).
2024/07/23 20:30:50 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2024/07/23 20:30:50 wazuh-syscheckd: INFO: FIM sync module started.
2024/07/23 20:30:51 wazuh-monitord: INFO: Started (pid: 59309).
2024/07/23 20:30:52 wazuh-logcollector: INFO: (9203): Monitoring journal entries.
2024/07/23 20:30:52 wazuh-modulesd:router: INFO: Loaded router module.
2024/07/23 20:30:52 wazuh-modulesd:content_manager: INFO: Loaded content_manager module.
2024/07/23 20:30:52 wazuh-modulesd: INFO: Started (pid: 59361).
2024/07/23 20:30:52 wazuh-modulesd:agent-upgrade: INFO: (8153): Module Agent Upgrade started.
2024/07/23 20:30:52 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
2024/07/23 20:30:52 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
2024/07/23 20:30:52 sca: INFO: Module started.
2024/07/23 20:30:52 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:30:52 wazuh-modulesd:vulnerability-scanner: INFO: Starting vulnerability_scanner module.
2024/07/23 20:30:52 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/23 20:30:52 wazuh-modulesd:task-manager: INFO: (8200): Module Task Manager started.
2024/07/23 20:30:52 wazuh-modulesd:router: INFO: Starting router module.
2024/07/23 20:30:52 wazuh-modulesd:content_manager: INFO: Starting content_manager module.
2024/07/23 20:30:52 wazuh-modulesd:database: INFO: Module started.
2024/07/23 20:30:52 wazuh-modulesd:download: INFO: Module started.
2024/07/23 20:30:52 wazuh-modulesd:control: INFO: Starting control thread.
2024/07/23 20:30:52 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:30:52 wazuh-modulesd:syscollector: INFO: Module started.
2024/07/23 20:30:52 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/07/23 20:30:52 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2024/07/23 20:30:52 indexer-connector: WARNING: IndexerConnector initialization failed for index 'wazuh-states-vulnerabilities-wazuh-server', retrying until the connection is successful.
2024/07/23 20:30:52 wazuh-modulesd:vulnerability-scanner: INFO: Vulnerability scanner module started.
2024/07/23 20:30:57 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_ubuntu22-04.yml'
2024/07/23 20:30:57 sca: INFO: Security Configuration Assessment scan finished. Duration: 5 seconds.
2024/07/23 20:30:59 indexer-connector: INFO: IndexerConnector initialized successfully for index: wazuh-states-vulnerabilities-wazuh-server.
2024/07/23 20:31:14 wazuh-modulesd:vulnerability-scanner: INFO: Initiating update feed process.
2024/07/23 20:31:39 rootcheck: INFO: Ending rootcheck scan.
2024/07/23 20:34:15 wazuh-modulesd:vulnerability-scanner: INFO: Triggered a re-scan after content update.
2024/07/23 20:34:15 wazuh-modulesd:vulnerability-scanner: INFO: Feed update process completed.
2024/07/23 20:42:35 wazuh-authd: INFO: New connection from 192.168.1.78
2024/07/23 20:42:35 wazuh-authd: INFO: Received request for a new agent (wazuh-agent) from: 192.168.1.78
2024/07/23 20:42:35 wazuh-authd: INFO: Agent key generated for 'wazuh-agent' (requested by any)
2024/07/23 20:42:39 wazuh-remoted: INFO: (1409): Authentication file changed. Updating.
2024/07/23 20:42:39 wazuh-remoted: INFO: (1410): Reading authentication keys file.
root@wazuh-server:/home/wazuh# 
```

</details>

<details><summary>Agent's log</summary>

```console
root@wazuh-agent:/home/wazuh# cat /var/ossec/logs/ossec.log 
2024/07/23 20:42:35 wazuh-execd: INFO: Started (pid: 5033).
2024/07/23 20:42:35 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2024/07/23 20:42:35 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2024/07/23 20:42:35 wazuh-agentd: INFO: Version detected -> Linux |wazuh-agent |6.1.0-23-amd64 |#1 SMP PREEMPT_DYNAMIC Debian 6.1.99-1 (2024-07-15) |x86_64 [Debian GNU/Linux|debian: 12 (bookworm)] - Wazuh v4.9.0
2024/07/23 20:42:35 wazuh-agentd: INFO: Started (pid: 5041).
2024/07/23 20:42:35 wazuh-agentd: INFO: Requesting a key from server: 192.168.1.77
2024/07/23 20:42:35 wazuh-agentd: INFO: No authentication password provided
2024/07/23 20:42:35 wazuh-agentd: INFO: Using agent name as: wazuh-agent
2024/07/23 20:42:35 wazuh-agentd: INFO: Waiting for server reply
2024/07/23 20:42:35 wazuh-agentd: INFO: Valid key received
2024/07/23 20:42:35 wazuh-agentd: INFO: Waiting 20 seconds before server connection
2024/07/23 20:42:36 wazuh-syscheckd: INFO: Started (pid: 5054).
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/boot', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6000): Starting daemon...
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2024/07/23 20:42:36 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
2024/07/23 20:42:37 wazuh-logcollector: INFO: Monitoring output of command(360): df -P
2024/07/23 20:42:37 wazuh-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
2024/07/23 20:42:37 wazuh-logcollector: INFO: Monitoring full output of command(360): last -n 20
2024/07/23 20:42:37 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
2024/07/23 20:42:37 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/log/dpkg.log'.
2024/07/23 20:42:37 wazuh-logcollector: INFO: Started (pid: 5067).
2024/07/23 20:42:38 wazuh-modulesd: INFO: Started (pid: 5085).
2024/07/23 20:42:38 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
2024/07/23 20:42:38 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
2024/07/23 20:42:38 sca: INFO: Module started.
2024/07/23 20:42:38 wazuh-modulesd:agent-upgrade: INFO: (8153): Module Agent Upgrade started.
2024/07/23 20:42:38 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:42:38 wazuh-modulesd:control: INFO: Starting control thread.
2024/07/23 20:42:38 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/23 20:42:38 wazuh-modulesd:syscollector: INFO: Module started.
2024/07/23 20:42:38 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/07/23 20:42:38 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:42:38 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2024/07/23 20:42:39 wazuh-logcollector: INFO: (9203): Monitoring journal entries.
2024/07/23 20:42:55 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2024/07/23 20:42:55 wazuh-agentd: INFO: Using AES as encryption method.
2024/07/23 20:42:55 wazuh-agentd: INFO: Trying to connect to server ([192.168.1.77]:1514/tcp).
2024/07/23 20:42:55 wazuh-agentd: INFO: (4102): Connected to the server ([192.168.1.77]:1514/tcp).
2024/07/23 20:42:55 wazuh-agentd: INFO: Agent is restarting due to shared configuration changes.
2024/07/23 20:42:55 wazuh-modulesd:syscollector: INFO: Stop received for Syscollector.
2024/07/23 20:42:56 wazuh-syscheckd: INFO: Agent is now online. Process unlocked, continuing...
2024/07/23 20:42:56 rootcheck: INFO: Starting rootcheck scan.
2024/07/23 20:42:57 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:42:57 sca: INFO: Security Configuration Assessment scan finished. Duration: 19 seconds.
2024/07/23 20:42:58 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2024/07/23 20:42:58 wazuh-syscheckd: INFO: FIM sync module started.
2024/07/23 20:42:58 wazuh-modulesd: INFO: Agent is now online. Process unlocked, continuing...
2024/07/23 20:42:58 wazuh-modulesd:syscollector: INFO: Module finished.
2024/07/23 20:42:58 wazuh-logcollector: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:42:59 wazuh-syscheckd: INFO: (1756): Shutdown received. Releasing resources.
2024/07/23 20:42:59 wazuh-syscheckd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:42:59 wazuh-agentd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:42:59 wazuh-execd: INFO: (1314): Shutdown received. Deleting responses.
2024/07/23 20:42:59 wazuh-execd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/07/23 20:43:00 wazuh-execd: INFO: Started (pid: 5976).
2024/07/23 20:43:01 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2024/07/23 20:43:01 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2024/07/23 20:43:01 wazuh-agentd: INFO: Version detected -> Linux |wazuh-agent |6.1.0-23-amd64 |#1 SMP PREEMPT_DYNAMIC Debian 6.1.99-1 (2024-07-15) |x86_64 [Debian GNU/Linux|debian: 12 (bookworm)] - Wazuh v4.9.0
2024/07/23 20:43:01 wazuh-agentd: INFO: Started (pid: 5987).
2024/07/23 20:43:01 wazuh-agentd: INFO: Using AES as encryption method.
2024/07/23 20:43:01 wazuh-agentd: INFO: Trying to connect to server ([192.168.1.77]:1514/tcp).
2024/07/23 20:43:01 wazuh-agentd: INFO: (4102): Connected to the server ([192.168.1.77]:1514/tcp).
2024/07/23 20:43:02 wazuh-syscheckd: INFO: Started (pid: 6001).
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/boot', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/bin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6003): Monitoring path: '/usr/sbin', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | scheduled'.
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mtab'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/hosts.deny'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/mail/statistics'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random-seed'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/random.seed'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/adjtime'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/httpd/logs'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/utmpx'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/wtmpx'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/cups/certs'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/dumpdates'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6206): Ignore 'file' entry '/etc/svc/volatile'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6207): Ignore 'file' sregex '.log$|.swp$'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6004): No diff for file: '/etc/ssl/private.key'
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6000): Starting daemon...
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6010): File integrity monitoring scan frequency: 43200 seconds
2024/07/23 20:43:02 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
2024/07/23 20:43:02 rootcheck: INFO: Starting rootcheck scan.
2024/07/23 20:43:03 wazuh-logcollector: INFO: Monitoring output of command(360): df -P
2024/07/23 20:43:03 wazuh-logcollector: INFO: Monitoring full output of command(360): netstat -tulpn | sed 's/\([[:alnum:]]\+\)\ \+[[:digit:]]\+\ \+[[:digit:]]\+\ \+\(.*\):\([[:digit:]]*\)\ \+\([0-9\.\:\*]\+\).\+\ \([[:digit:]]*\/[[:alnum:]\-]*\).*/\1 \2 == \3 == \4 \5/' | sort -k 4 -g | sed 's/ == \(.*\) ==/:\1/' | sed 1,2d
2024/07/23 20:43:03 wazuh-logcollector: INFO: Monitoring full output of command(360): last -n 20
2024/07/23 20:43:03 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/ossec/logs/active-responses.log'.
2024/07/23 20:43:03 wazuh-logcollector: INFO: (1950): Analyzing file: '/var/log/dpkg.log'.
2024/07/23 20:43:03 wazuh-logcollector: INFO: Started (pid: 6018).
2024/07/23 20:43:04 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2024/07/23 20:43:04 wazuh-syscheckd: INFO: FIM sync module started.
2024/07/23 20:43:04 wazuh-modulesd: INFO: Started (pid: 6068).
2024/07/23 20:43:04 wazuh-modulesd:ciscat: INFO: Module disabled. Exiting...
2024/07/23 20:43:04 sca: INFO: Module started.
2024/07/23 20:43:04 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:43:04 wazuh-modulesd:osquery: INFO: Module disabled. Exiting...
2024/07/23 20:43:04 wazuh-modulesd:agent-upgrade: INFO: (8153): Module Agent Upgrade started.
2024/07/23 20:43:04 wazuh-modulesd:control: INFO: Starting control thread.
2024/07/23 20:43:04 sca: INFO: Starting Security Configuration Assessment scan.
2024/07/23 20:43:04 wazuh-modulesd:syscollector: INFO: Module started.
2024/07/23 20:43:04 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2024/07/23 20:43:04 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:43:05 wazuh-modulesd:syscollector: INFO: Evaluation finished.
2024/07/23 20:43:05 wazuh-logcollector: INFO: (9203): Monitoring journal entries.
2024/07/23 20:43:09 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_debian12.yml'
2024/07/23 20:43:09 sca: INFO: Security Configuration Assessment scan finished. Duration: 5 seconds.

```
</details>

**Comment by ncvicchi** [comment]
## AWS Configuration

### Prerequisite: [Configure AWS Config](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-setup-prereqs.html#securityhub-prereq-config)

### Enable Security Hub manually

### Integrating Security Hub with EventBridge

#### [Creating an event rule for automatically sent findings](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-all-findings.html#securityhub-cwe-all-findings-predefined-pattern)

#### [Defining a rule for using custom actions to send findings and insight results](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-custom-actions.html#securityhub-cwe-define-rule)

### Amazon Simple Queue Service
#### Set up a Standard type SQS Queue with the default configurations

### Amazon S3 and Event Notifications

## Wazuh Configuration

## Authentication

## Troubleshooting
### Checking if the module is running
### Enabling debug mode
### Checking if logs are being processed

**Comment by ncvicchi** [comment]
### NOTE:

The AWS environment configuration has been a great challenge to overcome since the documentation expects a lot of prior knowledge regarding AWS administration, features, and concepts.
It also requires several permissions to be granted, which appear eventually when every step is being performed.

This creates a great deal of delays since a lot of referred documentation needs to be read, requests to other teams are solicited, and errors occur that are not straightforward to analyze.

## AWS Configuration :red_circle: 

### Prerequisite: [Configure AWS Config](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-setup-prereqs.html#securityhub-prereq-config) :yellow_circle: 

These prerequisites presented a challenge to be obtained. It indicates that a root account should be used (which was not possible). Then it required several permissions to be granted, and yet it is not working properly after that.

### Enable Security Hub :red_circle: 

The documentation presents 2 different ways to configure Security Hub, but it is not that simple for a non-AWS admin to determine which one to use. It was eventually used the manual option.

Although AWS Config permissions and the resource recording feature were supposedly granted, these errors were present in the Security Standard step:

![image](https://github.com/user-attachments/assets/b1e1aacb-d2bd-4405-9fb8-b85a3a8dde48)

It was indicated to continue anyway (the DevOps team did the standards assignment).

### AWS managed policies attachment to the IAM identity :yellow_circle: 

Every step uses concepts not necessary known, and requires time to read AWS documentation.
In this case, these 2 policies were required to be added. 

- AWSSecurityHubFullAccess
- AWSSecurityHubOrganizationsAccess

When trying to add them, once again permissions forbid to continue:

![image](https://github.com/user-attachments/assets/50faf56c-1520-4fb4-bb4b-5562e0e3af64)

![image](https://github.com/user-attachments/assets/9399e2de-523a-4059-b96a-8f7c6b32e55b)


Once again, the DevOps team had to grant them.


### Integrating Security Hub with EventBridge

#### [Creating an event rule for automatically sent findings](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-all-findings.html#securityhub-cwe-all-findings-predefined-pattern)

#### [Defining a rule for using custom actions to send findings and insight results](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cwe-custom-actions.html#securityhub-cwe-define-rule)

A note is included recommending to use of AWS central configuration, but it is not clear the purpose of such a note.

### Integrating Security Hub with EventBridge, assuming known by the user.

The documentation then presents EventBridge. It explains that there are 3 event types, but it is not clear if the 3 types should be tested, and the documentation specifies that for 2 of them AWS documentation should be followed. It is not clear the situation of the first type of event. It is also not clear the custom actions to be created for the later 2.

Once again, the documentation assumes knowledge on AWS Data Firehose. It also recommends Wazuh's Amazon WAF integration as example of configuration. It is not clear the relation between Amazon WAF (that it is mentioned by the first time) and the Security Hub configuration.

### Amazon Simple Queue Service :black_circle: 
#### Set up a Standard type SQS Queue with the default configurations :black_circle: 

### Amazon S3 and Event Notifications :black_circle: 

## Wazuh Configuration :black_circle: 

## Authentication :black_circle: 

## Troubleshooting :black_circle: 
### Checking if the module is running
### Enabling debug mode
### Checking if logs are being processed

**Comment by nico-stefani** [comment]
# Review

You are missing to open the corresponding issues for the problems you faced to complete the test.

According to exposed challenges, it could be:
* To the documentation side to simplify it or add more examples.
* To the E2E itself to provide more tools(permission required, configurations, etc)

And any other topic that you consider.

**Comment by ncvicchi** [comment]
Regarding the documentation simplification, if an issue is opened then it would change its status to failed, and it actually depends on the intended audiencie. I don't think I can take such decision.

Regarding the E2E itself, actually this issue requires a feedback where considerations were given regarding both the E2E design and the requisites that I assume would be necessary to better implement it.

**Comment by juliamagan** [comment]
As @ncvicchi points out, issue feedback is the right place to report and suggest improvements on E2E. However, the Wazuh documentation and these tests are intended to simulate the use cases that a Wazuh user may have, so if the documentation is not clear enough, an issue needs to be opened to improve it, and the test will be marked as failed.

**Comment by ncvicchi** [comment]
> As @ncvicchi points out, issue feedback is the right place to report and suggest improvements on E2E. However, the Wazuh documentation and these tests are intended to simulate the use cases that a Wazuh user may have, so if the documentation is not clear enough, an issue needs to be opened to improve it, and the test will be marked as failed.

Done @juliamagan 
[Improve AWS Security Hub Documentation #7588](https://github.com/wazuh/wazuh-documentation/issues/7588)

**Comment by rauldpm** [comment]
- Wazuh Manager discard regex: Skipped - Since events cannot be generated at will, this test was skipped
- Wazuh Agent Events & Dashboard: Skipped - Since events cannot be generated at will, this test was skipped

If events cannot be generated you should ask for help, but a test cannot be marked as skipped for that, also, this is no a decision that the tester can take

- Wazuh Manager discard regex: Skipped - Since configuration didn't worked, this test was skipped

I see the Wazuh configuration in the Wazuh manager as :green_circle: so this does not make sense, please expand this conclusion

**Comment by ncvicchi** [comment]
The decision not to continue with the test was based on the low quality of the documentation provided as well as the poorly designed E2E test (including the absence of adequate access rights and/or profiles to perform them), furthermore considering that this was the first time this test was being executed. 
Since all these situations prevented us from reaching the due date, we opted to require pyserver team with an already configured environment to probe Wazuh's functionalities the best we could. 
This decision was not taken by the tester, but in conjunction with the team leader.
It was also not possible to require help since a lot of delays were generated due to all these flaws, forcing us to work after hours, when nobody was available to ask for help.
If these tests are still necessary to be executed to close this issue before the E2E is redesigned and the documentation updated, please provide the steps to proceed, and the teams to be consulted, besides informing the necessary access rights an AWS account should have to perform them.

Regarding Wazuh Manager configuration, it was a mistake in the title of the table column. It has just been corrected.

**Comment by rauldpm** [comment]
> The decision not to continue with the test was based on the low quality of the documentation provided as well as the poorly designed E2E test (including the absence of adequate access rights and/or profiles to perform them), furthermore considering that this was the first time this test was being executed.

@ncvicchi the decision to continue or not the test is something that the release team should decide, not the testers, you should have elevated this so we could take action, like the issue opened to improve the documentation, maybe someone could have helped to generate events in AWS

> This decision was not taken by the tester, but in conjunction with the team leader.

Again, this is not a decision for the team leader to take, but the release team cc @cborla 

> It was also not possible to require help since a lot of delays were generated due to all these flaws, forcing us to work after hours, when nobody was available to ask for help.

Nobody (that I know of) asked you to work after hours, if someone asked you that, you should elevate it, in any case, you could have left a comment and pinged us (the release team) here and in Slack

> If these tests are still necessary to be executed to close this issue before the E2E is redesigned and the documentation updated, please provide the steps to proceed, and the teams to be consulted, besides informing the necessary access rights an AWS account should have to perform them.

This issue has been worked on for approximately five to six days, enough time to decide if we need to change something, we could not complete this test properly on time so we will need to re-test it in the next stage, closing this issue as incompleted with failures as ETA has been surpassed cc @wazuh/devel-qa-release
